### PR TITLE
docs: Configure Prettier for docs

### PR DESCRIPTION
**Pull request by Zamiell**
PR for #17504

### CODE REVIEW

File: CHANGELOG.md
Comment (nzakas) [inline CHANGELOG.md:1]: We should ignore this file because it is autogenerated.

File: docs/src/library/rule.md [hunk 1]
```diff
@@ -60,13 +60,13 @@ A rule has a:
...
-        hasSuggestions: false
-    }
+name: "getter-return",
+deprecated: false,
+description: 'Enforce `return` statements in getters.',
+categories: {
+recommended: true,
+fixable: false,
+hasSuggestions: false
+}
```
Comment (nzakas) [inline docs/src/library/rule.md:70]: Can we ignore files in the `/library` folder? These are showing macro examples and the indentation is useful to have.

File: docs/src/pages/rules.md
Comment (nzakas) [inline docs/src/pages/rules.md:1]: Can we ignore this file too?

**Review by nzakas — COMMENTED** [review]
Overall this looks pretty good to me. Aside from the few files we should ignore, it would also be helpful to add a precommit hook to run Prettier on these files.

**Comment by Zamiell** [comment]
not a fan of pre-commit hooks myself.
it is laggy for developers and produces pain, especially when making lots of nice, small commits.
a pre-commit hook is nice in the situation where one developer is pushing directly to a "main" branch without any other checks and balances.
but for ESLint, and any other relatively big project, most code is coming in to the project through pull requests, and a pull request will never be merged if CI isn't passing.
i find a much better workflow is to configure the repo settings.jon file for vscode to have it automatically summon prettier every time a file is saved.
this is essentially lag free and is the shortest possible distance between type-time and notice-error-time.
(noticing at commit-time with a pre-commit-hook is a midway point between noticing at PR-time with CI, but comes with an nasty lag cost)

**Comment by Zamiell** [comment]
I added the requested exceptions to the .prettierignore file: https://github.com/eslint/eslint/pull/17737/commits/625b360e820808c5e139e9e178e44be4e687863b

**Review by kecrily — COMMENTED** [review]
I'm not sure this is necessary, as of right now I think all these changes can be detected and fixed by the markdown-lint we are using

**Comment by Zamiell** [comment]
markdown-lint requires that a human make the changes, so that isn't really a viable solution. Furthermore, linters should be separate from formatters for the exact same reasons that ESLint itself is deprecating its formatting rules!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -10,10 +10,10 @@ This project adheres to the [Open JS Foundation Code of Conduct](https://eslint.
...
 Before filing an issue, please be sure to read the guidelines for what you're reporting:
 
-* [Report Bugs](https://eslint.org/docs/latest/contribute/report-bugs)
-* [Propose a New Rule](https://eslint.org/docs/latest/contribute/propose-new-rule)
-* [Propose a Rule Change](https://eslint.org/docs/latest/contribute/propose-rule-change)
-* [Request a Change](https://eslint.org/docs/latest/contribute/request-change)
+*   [Report Bugs](https://eslint.org/docs/latest/contribute/report-bugs)
+*   [Propose a New Rule](https://eslint.org/docs/latest/contribute/propose-new-rule)
+*   [Propose a Rule Change](https://eslint.org/docs/latest/contribute/propose-rule-change)
+*   [Request a Change](https://eslint.org/docs/latest/contribute/request-change)
```
Comment (mdjermanovic) [inline CONTRIBUTING.md:16]: These two extra spaces is a bug in Prettier? https://github.com/prettier/prettier/issues/5019

Comment (Zamiell) [reply CONTRIBUTING.md:16]: oh, can I fix it by forcing tab indentation to 2?

Comment (mdjermanovic) [reply CONTRIBUTING.md:16]: We'd prefer 4, perhaps we could wait to see if the bug might be fixed soon: https://github.com/prettier/prettier/pull/15526

**Comment by nzakas** [comment]
We already run precommit hooks for most ESLint files, so let's keep it consistent and do it for Prettier, too.

**Comment by Rec0iL99** [comment]
Ping @Zamiell

File: CONTRIBUTING.md (hunk 2 above)
Comment (Zamiell) [reply CONTRIBUTING.md:16]: for now i'll change to 2, it looks like that PR has been sitting for a while

**Comment by Zamiell** [comment]
Ok, I added the pre-commit hook and fix the list tab issue, the diff looks a lot cleaner now.

**Comment by Zamiell** [comment]
Looks like ESLint throws a spurious error:

```
/home/runner/work/eslint/eslint/prettier.config.mjs
  33:1  error  Parsing error: 'import' and 'export' may appear only with 'sourceType: module'
```

That's a bug in ESLint because an mjs file extension doesn't respect 'sourceType: module' at all. Does anyone know if there's already an issue opened about that?

**Comment by mdjermanovic** [comment]
> That's a bug in ESLint because an mjs file extension doesn't respect 'sourceType: module' at all. Does anyone know if there's already an issue opened about that?

It's not a bug, our eslint config for this project is just set to treat all files as CommonJS. Can the Prettier config file be a CommonJS `.js` file?

**Comment by Zamiell** [comment]
done

**Comment by kecrily** [comment]
@Zamiell Is it ready for review now?

**Comment by Zamiell** [comment]
@kecrily it has been ready to review since [this comment](https://github.com/eslint/eslint/pull/17737#issuecomment-1830029476)

**Review by kecrily — COMMENTED** [review]
There's a big problem with indent. This is caused by the prettier bug mentioned above. https://github.com/prettier/prettier/issues/5019

`tabWidth: 2` doesn't fix it, it just goes from one problem to another

File: package.json [hunk 3]
```diff
@@ -37,7 +38,7 @@
   },
   "lint-staged": {
     "*.js": "eslint --fix",
-    "*.md": "markdownlint --fix",
+    "*.md": "prettier --write && markdownlint --fix",
```
[the remark this answers is not in the document]
Comment (Zamiell) [reply package.json:41]: can you make it as a suggestion so i can click "commit suggestion"?

Comment (kecrily) [inline package.json:41]:
```suggestion
    "*.md": [
      "prettier --write",
      "markdownlint --fix"
    ],
   ```

[the remark this answers is not in the document]
Comment (kecrily) [reply package.json]: sure

**Pull request closed without merging**